src/components/encoder/Encoder.ts

```diff
@@ -0,0 +1,25 @@
+import type { Extra, PositionType } from '../../types';
+
+export interface Properties {
+  /** Whether a encoder supports ticks and/or degrees. */
+  ticksCountSupported: boolean;
+  angleDegreesSupported: boolean;
+}
```

Comment:
we're gonna use a new pattern here (sorry for leading you astray with Motor!) - we can just re-export the `.AsObject` type from the underlying proto

```suggestion
import pb from '../../gen/service/encoder/v1/encoder_pb';

export const EncoderProperties = pb.Properties.AsObject;
```

See for an example of this: https://github.com/viamrobotics/viam-typescript-sdk/pull/55/files#diff-1aa6d58266a33208d7c59494dad01b893ebe3bc15fe369667d7554cc929e0297R3

(dealers choice on whether you create a dedicated `./types` folder or just keep the definition here - we can figure out structuring later).